src/App.tsx

```diff
@@ -1,17 +1,150 @@
-import React from 'react';
-import './App.css';
-
+import React, { useState } from 'react';
+// import './App.scss';
+import cn from 'classnames';
 import users from './api/users';
+import todos from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './apptypes.defs';
+
+const preparedTodos: Todo[] = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
+
+const getUserByID = (userId: number) => {
+  return users.find(user => userId === user.id) || null;
+};
 
 const App: React.FC = () => {
+  const [todosList, setTodos] = useState([...preparedTodos]);
+  const [todoTitle, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+
+  const [hasTitleInputError, setHasTitleInputError] = useState(false);
+  const [hasUserInputError, setHasUserInputError] = useState(false);
+
+  const addtodo = (title: string, userId: number) => {
+    const newTodo: Todo = {
+      userId,
+      id: todosList.length + 1,
```

Comment:
> - `id` is a maximal `id` in the array + 1

not `list.length`
something like this:
`Math.max(...array.map((todo) => todo.id)) + 1`